build/source/engine/read_attrb.f90

```diff
@@ -97,70 +122,177 @@ USE get_ixname_module,only:get_ixAttr,get_ixType  ! access function to find inde
  ! build filename
  infile = trim(SETNGS_PATH)//trim(LOCAL_ATTRIBUTES)
  ! open file
- call file_open(trim(infile),unt,err,cmessage)
+ mode=nf90_NoWrite
+ call nc_file_open(trim(infile), mode, ncid, err, cmessage)
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
 
+ ! allocate space for GRU indices
+ allocate(gru_id(nGRU),gru_ix(nGRU),stat=err)
+ if(err/=0)then; err=20; message=trim(message)//'problem allocating space for GRU indices'; return; endif
+
+ ! allocate space for hru-gru mapping vectors
+ allocate(hru_id(nHRU),hru_ix(nHRU),hru2gru_id(nHRU),hru2gru_ixGRU(nHRU),hru2gru_ixHRU(nHRU), stat=err)
+ if(err/=0)then; err=20; message=trim(message)//'problem allocating space for gru-hru mapping vectors'; return; endif
 
  ! **********************************************************************************************
- ! (2) read local attributes
+ ! (2) read mapping vectors and populate mapping structures
  ! **********************************************************************************************
- ! ---------------------------------------------------------------------------------------------
- ! read attribute names
- ! ---------------------------------------------------------------------------------------------
- do iline=1,maxLines
-  ! (read through comment lines)
-  read(unt,'(a)',iostat=iend) temp  ! read a line of data
-  if(iend/=0)then; err=20; message=trim(message)//'got to end of file before found the format code'; return; endif
-  if (temp(1:1)=='!')cycle
-  ! (read in format string -- assume that the first non-comment line is the list of attribute names)
-  read(temp,'(a)')nameString  ! read in list of attribute names
-  exit
-  if(iLine==maxLines)then; err=20; message=trim(message)//'problem finding list of attribute names'; return; endif
- end do ! looping through lines
- ! ---------------------------------------------------------------------------------------------
- ! identify the type of each attribute
- ! ---------------------------------------------------------------------------------------------
- ! split the line into an array of words
- call split_line(nameString,attNames,err,cmessage)
- if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
- ! identify the number of attributes
- nAtt = size(attNames)
- ! allocate space for the variable type and index
+
+ ! read gru_id from netcdf file
+ err = nf90_inq_varid(ncid, "gruId", varid); if(err/=0)then; message=trim(message)//'problem finding gruId'; return; endif
```

Comment:
Yes. Opened issue #133.

Currently the NetCDF files assume specific names, and it is the responsibility of the user to convert their files to the summa naming conventions (e.g., using NCO).

It would be less work for the user if we provided a dictionary, that enables a user to relate their variable names to the summa variable names. We should *consider* this.